newrelic/resource_newrelic_insights_event_test.go

```diff
@@ -67,18 +68,32 @@ func testAccCheckNewRelicInsightsEventExists(n string, nrqls []string) resource.
 			return fmt.Errorf("no event ID is set")
 		}
 
-		client := testAccProvider.Meta().(*ProviderConfig).InsightsQueryClient
+		providerConfig := testAccProvider.Meta().(*ProviderConfig)
+		client := providerConfig.NewClient
+		accountID := providerConfig.AccountID
 
+		// Due to the asynchronous operation, we need to
+		// wait for the events to propagate.
+		time.Sleep(15 * time.Second)
+
+		var errs []string
 		for _, nrql := range nrqls {
-			resp, err := client.QueryEvents(nrql)
+			resp, err := client.Nrdb.Query(accountID, nrdb.Nrql(nrql))
 			if err != nil {
-				return err
+				errs = append(errs, err.Error())
 			}
+
 			if len(resp.Results) != 1 {
-				return errors.New("did not find Insights event")
+				errs = append(errs, fmt.Sprintf("[Error] Insights event not found (likely due to async issue) - query: %v", nrql))
 			}
 		}
 
+		if len(errs) > 0 {
+			fmt.Printf("test case TestAccNewRelicInsightsEvent_Basic failed (likely due to asynchronous)")
+
+			return fmt.Errorf("%v", strings.Join(errs, "\n"))
+		}
+
```

Comment:
We should remove this test since we do not expose the query method, there is no reason to test for it.  Testing that an event was inserted in an async pipeline is going to be problematic, and not add any additional information about the workings of the provider than what the status code return on the create provides.